File: python/7 kyu/alternate_capitalization.py
# 7 kyu Alternate capitalization
# https://www.codewars.com/kata/59cfc000aeb2844d16000075/train/python

def capitalize(s):
  lis = list(s)
  results_even = []
  results_odd = []
  
  for i in range(len(lis)):
    if i % 2 == 0:
      results_even.append(lis[i].upper())
    else:
      results_even.append(lis[i])

  for i in range(len(lis)):
    if i % 2 == 0:
      results_odd.append(lis[i])
    else:
      results_odd.append(lis[i].upper())
    
  return [''.join(results_even), ''.join(results_odd)]

# Best Solution
def capitalize(s):
    s = ''.join(c if i%2 else c.upper() for i,c in enumerate(s))
    return [s, s.swapcase()]

File: javascript/7_kyu/alternate_capitalization.js
// Alternate capitalization
// https://www.codewars.com/kata/59cfc000aeb2844d16000075/train/javascript

function capitalize(s) {
    var even_cap = [];
    var odd_cap = [];

    for (var i = 0; i < s.length; i++) {
        if (i % 2 == 0) {
            even_cap.push(s[i].toUpperCase());
            odd_cap.push(s[i]);
        } else {
            odd_cap.push(s[i].toUpperCase());
            even_cap.push(s[i]);
        }
    }

    return [even_cap.join(''), odd_cap.join('')];
}

console.log(capitalize("codewars"));